Expensify/App
# fix: Long filename handling in Preview

### Details

- Long filename was overflowing outside the container. Added a component that adds ellipses (...) in the middle of the string. 
- Updated `subtitle` prop to `Header` so that a custom component or string can be passed to `Header`

### Fixed Issues

$ https://github.com/Expensify/App/issues/6873

### Tests
1. Tested with long filenames on Desktop, Web, and Mobile Apps
2. Tested Fluid width to ensure ellipses show only if there isn't space for the whole name to show up.
3. Tested `ReportingTypingIndicator` as the component was updated.



### QA Steps
1. Launch the app and open any chat
2. Upload an attachment with a very long name
2. Open the attachment with the preview pane
3. Ensure that the filename should be visible with `...` and ensure that the extension should show up.



- [] Verify that no errors appear in the JS console

### Tested On


### Screenshots


#### Web

<img width="1254" alt="web-file-preview-long-text" src="https://user-images.githubusercontent.com/3069065/148654145-911822c7-93f1-4039-a82b-4ef4dbbf4285.png">

#### Mobile Web

<img width="327" alt="mweb-file-preview-long-text" src="https://user-images.githubusercontent.com/3069065/148654156-3928ab51-b85f-4917-bfd9-8786423a5d5c.png">

#### Desktop

<img width="1165" alt="desktop-file-preview-long-text" src="https://user-images.githubusercontent.com/3069065/148654158-8623b327-8382-4c76-bb92-2a9548c72c25.png">

#### iOS

<img width="339" alt="ios-file-preview-long-text" src="https://user-images.githubusercontent.com/3069065/148654163-f99ba843-fa7b-422a-b3b5-d8891cd7edf3.png">

#### Android

<img width="311" alt="android-file-preview-long-text" src="https://user-images.githubusercontent.com/3069065/148654171-54b6ba51-a30f-4a39-a084-ed308d28bd64.png">
